src/main/java/com/cloudbees/jenkins/plugins/sshagent/jna/JNRRemoteAgentStarter.java

```diff
@@ -55,4 +64,9 @@ public RemoteAgent call() throws Throwable {
         final Channel channel = Channel.current();
         return channel == null ? instance : channel.export(RemoteAgent.class, instance);
     }
+
+    @Override
+    public void checkRoles(RoleChecker arg0) throws SecurityException {
+        // TODO Auto-generated method stub
+    }
```

Comment:
No, never implement this method yourself. Extend `MasterToSlaveCallable` instead (assuming this is indeed a master → slave call).